### ISSUE TITLE
TypeError:forward() missing 1 required positinal argument: 'x'

### ISSUE THREAD

**Issue by lawsonX** [gistvision/bnas#3]
Hi! Thanks for your excellent contribution to the community! 
I faces a problem when I tried to simply run your code (search or inference ). 
It raised " TypeError:forward() missing 1 required positinal argument: 'x' " . This problem probably is in model_search.py line 136 & line 124. 
Can you take a look at it ?

**Comment by winstondcosta** [comment]
The error is due to the PyTorch version you are using. You need to make the following changes in the src/operations.py and the error will be resolved

class BinActive(torch.autograd.Function):
    '''
    Binarize the input activations and calculate the mean across channel dimension.
    '''
    @staticmethod
    def forward(self, input):
        self.save_for_backward(input)
        size = input.size()
        mean = torch.mean(input.abs(), 1, keepdim=True)
        input = input.sign()
        return input, mean

    @staticmethod
    def backward(self, grad_output, grad_output_mean):
        input, = self.saved_tensors
        grad_input = grad_output.clone()
        grad_input[input.ge(1)] = 0
        grad_input[input.le(-1)] = 0
        return grad_input

class BinReLUConvBN(nn.Module):

  def __init__(self, C_in, C_out, kernel_size, stride, padding, affine=True):
    super(BinReLUConvBN, self).__init__()
    self.bn = nn.BatchNorm2d(C_in, affine=affine)
    self.op = nn.Sequential(
      nn.Conv2d(C_in, C_out, kernel_size, stride=stride, padding=padding, bias=False),
      nn.ReLU(inplace=False),     
    )

  def forward(self, x):
     
    x = self.bn(x)
    x,_ = BinActive.apply(x)       
    return self.op(x)

class BinDilConv(nn.Module):
    
  def __init__(self, C_in, C_out, kernel_size, stride, padding, dilation, affine=True):
    super(BinDilConv, self).__init__()
    self.bn =  nn.BatchNorm2d(C_in, affine=affine)
    self.op = nn.Sequential(
      nn.Conv2d(C_in, C_out, kernel_size=kernel_size, stride=stride, padding=padding, dilation=dilation, bias=False),
     
       nn.ReLU(inplace=False),
      )

  def forward(self, x):
    x = self.bn(x)
    x, _ = BinActive.apply(x)
    return self.op(x)

**Comment by lawsonX** [comment]
> The error is due to the PyTorch version you are using. You need to make the following changes in the src/operations.py and the error will be resolved
> 
> class BinActive(torch.autograd.Function):
> '''
> Binarize the input activations and calculate the mean across channel dimension.
> '''
> @staticmethod
> def forward(self, input):
> self.save_for_backward(input)
> size = input.size()
> mean = torch.mean(input.abs(), 1, keepdim=True)
> input = input.sign()
> return input, mean
> 
> ```
> @staticmethod
> def backward(self, grad_output, grad_output_mean):
>     input, = self.saved_tensors
>     grad_input = grad_output.clone()
>     grad_input[input.ge(1)] = 0
>     grad_input[input.le(-1)] = 0
>     return grad_input
> ```
> 
> class BinReLUConvBN(nn.Module):
> 
> def **init**(self, C_in, C_out, kernel_size, stride, padding, affine=True):
> super(BinReLUConvBN, self).**init**()
> self.bn = nn.BatchNorm2d(C_in, affine=affine)
> self.op = nn.Sequential(
> nn.Conv2d(C_in, C_out, kernel_size, stride=stride, padding=padding, bias=False),
> nn.ReLU(inplace=False),
> )
> 
> def forward(self, x):
> 
> ```
> x = self.bn(x)
> x,_ = BinActive.apply(x)       
> return self.op(x)
> ```
> 
> class BinDilConv(nn.Module):
> 
> def **init**(self, C_in, C_out, kernel_size, stride, padding, dilation, affine=True):
> super(BinDilConv, self).**init**()
> self.bn = nn.BatchNorm2d(C_in, affine=affine)
> self.op = nn.Sequential(
> nn.Conv2d(C_in, C_out, kernel_size=kernel_size, stride=stride, padding=padding, dilation=dilation, bias=False),
> 
> ```
>    nn.ReLU(inplace=False),
>   )
> ```
> 
> def forward(self, x):
> x = self.bn(x)
> x, _ = BinActive.apply(x)
> return self.op(x)
